internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -0,0 +1,955 @@
+		EbsVolumeType: apiObject.EbsVolumeType,
+		InstanceSize:  apiObject.InstanceSize,
+	}
+}
+
+func dedicatedHwSpecToHwSpec(apiObject *admin.DedicatedHardwareSpec) *admin.HardwareSpec {
+	if apiObject == nil {
+		return nil
+	}
+	return &admin.HardwareSpec{
+		NodeCount:     apiObject.NodeCount,
+		DiskIOPS:      apiObject.DiskIOPS,
+		EbsVolumeType: apiObject.EbsVolumeType,
+		InstanceSize:  apiObject.InstanceSize,
+	}
+}
+
+func flattenAdvancedReplicationSpecRegionConfigSpec(apiObject *admin.DedicatedHardwareSpec, providerName string, tfMapObjects []any) []map[string]any {
+	if apiObject == nil {
+		return nil
+	}
+	var tfList []map[string]any
+
+	tfMap := map[string]any{}
+
+	if len(tfMapObjects) > 0 {
+		tfMapObject := tfMapObjects[0].(map[string]any)
+
+		if providerName == "AWS" {
+			if cast.ToInt64(apiObject.GetDiskIOPS()) > 0 {
+				tfMap["disk_iops"] = apiObject.GetDiskIOPS()
+			}
+			if v, ok := tfMapObject["ebs_volume_type"]; ok && v.(string) != "" {
+				tfMap["ebs_volume_type"] = apiObject.GetEbsVolumeType()
+			}
+		}
+		if _, ok := tfMapObject["node_count"]; ok {
+			tfMap["node_count"] = apiObject.GetNodeCount()
+		}
+		if v, ok := tfMapObject["instance_size"]; ok && v.(string) != "" {
+			tfMap["instance_size"] = apiObject.GetInstanceSize()
+			tfList = append(tfList, tfMap)
+		}
+	} else {
+		tfMap["disk_iops"] = apiObject.GetDiskIOPS()
+		tfMap["ebs_volume_type"] = apiObject.GetEbsVolumeType()
+		tfMap["node_count"] = apiObject.GetNodeCount()
+		tfMap["instance_size"] = apiObject.GetInstanceSize()
+		tfList = append(tfList, tfMap)
+	}
+	return tfList
+}
+
+func flattenAdvancedReplicationSpecAutoScaling(apiObject *admin.AdvancedAutoScalingSettings) []map[string]any {
+	if apiObject == nil {
+		return nil
+	}
+	var tfList []map[string]any
+	tfMap := map[string]any{}
+	if apiObject.DiskGB != nil {
+		tfMap["disk_gb_enabled"] = apiObject.DiskGB.GetEnabled()
+	}
+	if apiObject.Compute != nil {
+		tfMap["compute_enabled"] = apiObject.Compute.GetEnabled()
+		tfMap["compute_scale_down_enabled"] = apiObject.Compute.GetScaleDownEnabled()
+		tfMap["compute_min_instance_size"] = apiObject.Compute.GetMinInstanceSize()
+		tfMap["compute_max_instance_size"] = apiObject.Compute.GetMaxInstanceSize()
+	}
+	tfList = append(tfList, tfMap)
+	return tfList
+}
+
+func getAdvancedClusterContainerID(containers []admin.CloudProviderContainer, cluster *admin.CloudRegionConfig) string {
+	if len(containers) == 0 {
+		return ""
+	}
+	for i := range containers {
+		if cluster.GetProviderName() == constant.GCP {
+			return containers[i].GetId()
+		}
+		if containers[i].GetProviderName() == cluster.GetProviderName() &&
+			containers[i].GetRegion() == cluster.GetRegionName() || // For Azure
+			containers[i].GetRegionName() == cluster.GetRegionName() { // For AWS
+			return containers[i].GetId()
+		}
+	}
+	return ""
+}
+
+func expandProcessArgs(d *schema.ResourceData, p map[string]any) *admin.ClusterDescriptionProcessArgs {
```

Comment:
nit: any reason why we return a pointer here? Asking because in the usages the value is being accessed directly.

Reply:
thx, changed to make it more explicit that a value is always returned